tinygrad/renderer/cstyle.py

```diff
@@ -328,7 +328,9 @@ class WGSLLanguage(CStyleLanguage):
   barrier="workgroupBarrier();"
   generic_var_prefix = "var "
   external_local_bufs = True
-  code_for_op = { **CStyleLanguage().code_for_op, BinaryOps.CMPLT: lambda x,y,dtype: f"f32({x}<{y})", TernaryOps.MULACC: lambda x,y,z,dtype: f"fma({x},{y},{z})", TernaryOps.WHERE: lambda a,b,c,dtype: f"select({c},{b},{a}!=0.)" }
+  code_for_op = { **CStyleLanguage().code_for_op, TernaryOps.MULACC: lambda x,y,z,dtype: f"fma({x},{y},{z})", TernaryOps.WHERE: lambda a,b,c,dtype:
+  f"select({c},{b},f32({a})!=0.)", **{opcode: lambda a,b,dtype,op=op: f"bool(i32({a}){op}i32({b}))" if dtype == dtypes.bool else f"({a}{op}{b})"
```

Comment:
I don't think this is right with the bool casting. This should be fixed in a refactor, not with hacks

Reply:
the cpmplt f32 hack we had is also a hack, but ok i'm putting that back and will solve this in a nice way as a separate refactor. So, this pr will only include the custom for removal.